experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -77,41 +77,37 @@ var contextGVK schema.GroupVersionKind = schema.GroupVersionKind{
 	Kind:    "Context",
 }
 
-func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *ExpanderReconciler) updatePlanStatus(ctx context.Context, plancr *compositionv1alpha1.Plan, newStatus *compositionv1alpha1.PlanStatus) {
 	logger := log.FromContext(ctx)
 	logger = logger.WithName(r.Composition.Name).WithName(r.InputGVK.Group)
-
-	logger.Info("Got Input API for expansion", "request", req)
-
-	inputcr := unstructured.Unstructured{}
-	inputcr.SetGroupVersionKind(r.InputGVK)
-	if err := r.Get(ctx, req.NamespacedName, &inputcr); err != nil {
-		logger.Error(err, "unable to fetch Input API Object")
-		// we'll ignore not-found errors, since they can't be fixed by an immediate
-		// requeue (we'll need to wait for a new notification), and we can get them
-		// on deleted requests.
-		return ctrl.Result{}, client.IgnoreNotFound(err)
-	}
-
-	loggerCR := logger.WithName(inputcr.GetName())
-
-	// Grab the latest composition
-	// TODO(barni@) - Decide how we want the latest composition changes are to be applied.
-	var compositionCR compositionv1alpha1.Composition
-	if err := r.Client.Get(ctx, r.Composition, &compositionCR); err != nil {
-		if client.IgnoreNotFound(err) != nil {
-			logger.Error(err, "Unable to fetch Composition Object")
-			return ctrl.Result{}, err
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		nn := types.NamespacedName{Namespace: plancr.Namespace, Name: plancr.Name}
+		err := r.Client.Get(ctx, nn, plancr)
+		if err != nil {
+			return err
 		}
+		plancr.Status = *newStatus.DeepCopy()
+		return r.Client.Status().Update(ctx, plancr)
+	})
+	if err != nil {
+		logger.Error(err, "unable to update Plan status", "name", plancr.Name, "namespace", plancr.Namespace)
 	}
+}
 
-	// Associate a plan object with this input CR
+func (r *ExpanderReconciler) getPlanForInputCR(ctx context.Context, inputcr *unstructured.Unstructured) (types.NamespacedName,
+	*compositionv1alpha1.Plan, error) {
 	var plancr compositionv1alpha1.Plan
-	planNN := types.NamespacedName{Name: r.InputGVR.Resource + "-" + inputcr.GetName(), Namespace: inputcr.GetNamespace()}
+	logger := log.FromContext(ctx)
+	logger = logger.WithName(r.Composition.Name).WithName(r.InputGVK.Group).WithName(inputcr.GetName())
```

Comment:
Q: do the names replace each other?

Reply:
I have seen it append:

`compositionName.Group.inputcrName : log here `